walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,111 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.HashSet;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    Collection<TimeRange> availableTimeRanges = new ArrayList<TimeRange> ();
+
+    Collection<String> requestAttendees = request.getAttendees();
+    Collection<String> requestOptionalAttendees = request.getOptionalAttendees();
+    Set<String> requestAttendeesSet = new HashSet<> ();
+    Set<String> requestOptionalAttendeesSet = new HashSet<> ();
+    requestOptionalAttendeesSet.addAll(requestOptionalAttendees);
+    requestAttendeesSet.addAll(requestAttendees);
+    requestAttendeesSet.addAll(requestOptionalAttendees);
+    int requestDuration = (int) request.getDuration();
+    
+    Collection<TimeRange> allEventsTimeRanges = new ArrayList<TimeRange> ();
+    for (Iterator currEvent = events.iterator(); currEvent.hasNext(); ){ 
+      allEventsTimeRanges.add((TimeRange) ((Event) currEvent.next()).getWhen());
+    }
+
+    int startTime = 0;
+    int endTime = 0;
+    TimeRange tempTimeRange = TimeRange.fromStartEnd(startTime, endTime, true);
+    while (startTime <= 1439 && endTime <= 1439) {
+      if (checkOverlapMultipleTimeRanges(tempTimeRange, allEventsTimeRanges)) {
+        //don't add if the start=end but the timerange overlaps
+        //timerange has duration zero and should just be updated
```

Comment:
nit: comments should start with a space, and should be full sentences (capitalization and periods).

Reply:
Here and throughout plz!